include/picongpu/simulation/stage/SynchrotronRadiation.hpp

```diff
@@ -0,0 +1,279 @@
+         */
+        template<typename T_Number>
+        static T_Number integrateAsExponential(
+            T_Number const xLeft,
+            T_Number const xMiddle,
+            T_Number const xRight,
+            T_Number const yLeft,
+            T_Number const yMiddle)
+        {
+            if constexpr(picongpu::particles::synchrotron::T_Debug)
+            {
+                if(xLeft == xMiddle)
+                {
+                    throw "xLeft and xMiddle cannot be the same.";
+                }
+                if(yLeft <= 0 || yMiddle <= 0)
+                {
+                    throw "yLeft and yMiddle must be positive.";
+                }
+                if(yLeft == yMiddle)
+                {
+                    throw "yLeft and yMiddle must not be equal.";
+                }
+            }
+
+            // fitting function: y = a * e^(b * x)
+            T_Number const b = (math::log(yMiddle) - math::log(yLeft)) / (xMiddle - xLeft);
+            T_Number const a = yLeft / math::exp(b * xLeft);
+
+            T_Number const integral = (a / b) * (math::exp(b * xRight) - math::exp(b * xLeft));
+            return integral;
+        }
+
+        /// @todo insure intervals consistent
+        /** compute first synchrotorn function. See paper: "Extended particle-in-cell schemes for physics in
+         * ultrastrong laser fields: Review and developments" by A. Gonoskov et.Al.
+         *
+         * @param zq
+         *
+         * @returns zq * (integral of 2nd kind bessel function from zq to ~infinity)
+         */
+        static float_64 firstSynchrotronFunction(float_64 const zq)
+        {
+            using namespace picongpu::particles::synchrotron::params; // for "FirstSynchrotronFunctionParams" struct
+
+            float_64 const log_start = std::log2(zq); // from zq to FirstSynchrotronFunctionParams::logEnd
+            float_64 const log_step = (FirstSynchrotronFunctionParams::logEnd - log_start)
+                / (FirstSynchrotronFunctionParams::numberSamplePoints - 1);
+
+            float_64 integral = 0;
+
+            float_64 xLeft;
+            float_64 xRight = zq;
+
+            for(unsigned int i = 0; i < FirstSynchrotronFunctionParams::numberSamplePoints - 1; ++i)
+            {
+                xLeft = xRight;
+                xRight = math::pow(2., log_start + log_step * (i + 1));
+                float_64 xMiddle = (xLeft + xRight) / 2.0;
+
+                // try and catch errors in the bessel function
+                try
+                {
+                    float_64 yLeft = std::cyl_bessel_k(5.0 / 3.0, xLeft);
+                    float_64 yMiddle = std::cyl_bessel_k(5.0 / 3.0, xMiddle);
+                    integral += integrateAsExponential(
+                        xLeft,
+                        xMiddle,
+                        xRight,
+                        yLeft,
+                        yMiddle); // computes the integral over one interval: [xLeft, xRight] using the exponential
```

Comment:
please move the comment above the line you like to describe and use multi-line comments.

Reply:
done